### DISCUSSION TITLE
Fastlane set submit_for_review  & automatic_release async and create a log file

### DISCUSSION THREAD

**Discussion by EArminjon** [fastlane/fastlane#18637] [Ideas] [answered]
hi,

I used Fastlane to manage around 20 apps which need to be deployed all the time all together (same day for example).
Actually I've a script which take these apps one buy one and run for each the Android and IOS lanes. It take a very long time because of the await from automatic_release from ios :

In the IOS Fastlane stages., when used upload_to_app_store with "automatic_release" at "true", fastlane while loop to await until testflight is ready to be pushed for production. It could take 20minutes... (20minutes X 20 apps...)

Example of the problem :

```
[11:29:22]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
[11:29:38]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
[11:29:53]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
....
continues)
[11:32:26]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
[11:32:44]: Waiting for App Store Connect to finish processing the new build (4.1.3 - 24) for IOS
[11:33:00]: Waiting for App Store Connect to finish processing the new build (4.1.3 - 24) for IOS
....
[11:40::00]: Waiting for App Store Connect to finish processing the new build (4.1.3 - 24) for IOS
```

Is it possible to make this asynchronous and create a log file at the end or something like that ? With that, my script could continue to run and I will gain a huge time improvement !!

I maybe should use docker for each deployment process, but i didn't think it's a good solution to launch 20 containers at same times... Maybe a cloud base deployment but I'm afraid about the cost for 20 apps with weekly update...

I couldn't use parallels execution because of xcode which not support parallels build as I know.

If someone have a better solution, I'm open !

**Comment by lucgrabowski** [comment, marked as answer]
Hi, instead of creating a log file, you could split uploading and submitting for review into 2 phases: 
1. build all the apps and upload them
2. submit all the apps for review with automatic_release flag.

If you use dotenv files (https://docs.fastlane.tools/advanced/other/), it should be fairly easy to iterate over them twice.

**Reply by EArminjon** [reply]
Hi, thanks for your reply ! 

Oh it's possible :) ? 

Actually i do that to upload : 
```
 upload_to_app_store(
        copyright: '2021',
        skip_screenshots: true,
        skip_metadata: true,
        #submit_for_review: true,
        submission_information: {
              add_id_info_limits_tracking: true,
              add_id_info_serves_ads: true,
              add_id_info_tracks_action: true,
              add_id_info_tracks_install: true,
              add_id_info_uses_idfa: true,
              content_rights_has_rights: true,
              content_rights_contains_third_party_content: true,
              export_compliance_platform: 'ios',
              export_compliance_compliance_required: false,
              export_compliance_encryption_updated: false,
              export_compliance_app_type: nil,
              export_compliance_uses_encryption: false,
              export_compliance_is_exempt: false,
              export_compliance_contains_third_party_cryptography: false,
              export_compliance_contains_proprietary_cryptography: false,
              export_compliance_available_on_french_store: true,
            },
    )
```

So if i well understand, if I let commented "submit_for_review" and "automatic_release" and then call later the following it will work :) ?

```
upload_to_app_store(
   submit_for_review: true,
   automatic_release: true,
),
```

**Reply by lucgrabowski** [reply]
When submitting the build for review, you should use 
`skip_binary_upload`,  
`submit_for_review`, 
`automatic_release`
and `submission_information`
If you're not interested in uploading metadata and screenshots, you can of course use `skip_metadata` and `skip_screenshots`
Other options that you can use are described in https://docs.fastlane.tools/actions/upload_to_app_store/

**Reply by EArminjon** [reply]
Perfect :) ! 

I will implement that at the end of my deploy script, Thank you !
